File: frontend/src/app/types/dto/person/related/skill/soft/person-soft-skill-dto.ts
import { SoftSkillDto } from "../../../../skill/soft/soft-skill-dto";
import { PersonSoftSkillDtoId } from "../../reference/skill/person-soft-skill-dto-id";

export interface PersonSoftSkillDto {
    id: PersonSoftSkillDtoId; 
    softSkill: SoftSkillDto;
}

File: backend/src/main/java/ibee/webapp/todo_app/features/person/related/skill/dto/soft/PersonSoftSkillDto.java
package ibee.webapp.todo_app.features.person.related.skill.dto.soft;

import ibee.webapp.todo_app.dto.skills.hard.soft.SoftSkillDto;
import ibee.webapp.todo_app.features.person.related.referenceIds.skill.soft.PersonSoftSkillDtoId;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;

public record PersonSoftSkillDto(

    @NotNull
    @Valid
    SoftSkillDto softSkill,

    @Valid
    PersonSoftSkillDtoId id
) {

}
